isaac/controllers.md

```diff
@@ -0,0 +1,255 @@
+
+TODO: actual ART-report link
+
+In our [technical report](https://www.dropbox.com/scl/fi/84mxa8jb7muiar76twg16/art_controllers-9.pdf?rlkey=xcaj31k59riay7cvqwwi73s13&dl=0), we define rules and procedures for creating, updating, transferring, and tracking the state of tagged objects, and prove that our rules maintain safety properties including _causal resource history_ and _consistent controller tags_.
+In this blog post, we explain what properties we are trying to maintain, why we think these properties are important, and how they fit into the larger Anoma ecosystem. 
+We will touch on some of the machinery under the controller tag operations, but leave most of those details to the [technical report](https://www.dropbox.com/scl/fi/84mxa8jb7muiar76twg16/art_controllers-9.pdf?rlkey=xcaj31k59riay7cvqwwi73s13&dl=0).
+
+## System Model
+### Resources
+
+TODO: for the blog post, this section could be shorter somehow. 
+
+The digital objects we use in Anoma are called [resources](https://zenodo.org/doi/10.5281/zenodo.10498990).
+We can encode extremely general mutable state with resources, but resources themselves are fairly simple.
+Resources have very limited mutable state: they are _not yet created_ by default, can transition to _created_, and then to _consumed_.
+However, each resource can carry arbitrary immutable data, specified by the resource's unique ID (which can be a hash).
+
+Each controller's state carries cryptographic accumulators (e.g. [Merkle roots](https://doi.org/10.1007/3-540-48184-2_32)) representing the set of resources created by transactions ordered by this controller and the set of resources consumed.
+If a resource is neither created nor consumed, it is _not yet created_.
+As part of their immutable data, resources can have complex proof obligations (which we call _resource logics_) determining when they can be created or consumed, and these may depend on the state of other resources.
+A resource logic can for example, specify exactly what programs can consume this resource: it would require a proof that the resources created are precisely the outputs of running a particular program with the consumed resources as inputs.
+Such a proof might be as simple as a full execution trace of the program, or as complex as a [zero-knowledge proof](https://eprint.iacr.org/2021/370).
+Through these logics, resources can encode fairly arbitrary state, not limited to scalar registers or tokens, while still allowing [ZKP-style confidential transactions](https://zenodo.org/doi/10.5281/zenodo.10498990).
+Crucially, resource logics can also constrain _where a resource can be transferred_: not all resources will be able to be transferred to all controllers. 
+
+### Controller Tags
+Each resource's _controller tag_ is part of this immutable data.
```

Comment:
Is the controller tag information encoded in $r.label$ ? or is there another field that needs to be added to the resource to include:

- The creating controller
- The terminal controller
- A dependency graph (DAG) detailing which controllers may have corrupted the resource if they were Byzantine